lib/filesystem.nix

```diff
@@ -42,4 +44,24 @@
               type = (builtins.readDir parent).${base} or null;
           in file == /. || type == "directory";
     in go (if isDir then file else parent);
+
+  # filterMapDir :  filter : ( name : String -> type : String -> Bool )
+  #              -> f : (name : String -> type : String -> a )
+  #              -> path : Path
+  #              -> { path : Path = a }
```

Comment:
This type is very confusing with all these names, I'd stick to what other functions do instead